sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/BulkDeviceIdentityLifecycleSamples.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Iot.Hub.Service.Models;
+
+namespace Azure.Iot.Hub.Service.Samples
+{
+    /// <summary>
+    /// This sample goes through the lifecycle of multiple Device Identities for multiple devices. It utilizes
+    /// bulk APIs that allow for creating, updating, and deleting of up to 100 device identities at a time.
+    /// There are also bulk APIs for updating twins on up to 100 devices at time
+    /// </summary>
+    internal class BulkDeviceIdentityLifecycleSamples
+    {
+        public readonly IoTHubServiceClient IoTHubServiceClient;
+        public const int MaxRandomValue = 200;
+        public readonly Random Random = new Random();
+
+        // Can be 1 to 100, configures how many devices to create/update/delete per service API call
+        public const int BulkCount = 20;
+
+        public BulkDeviceIdentityLifecycleSamples(IoTHubServiceClient client)
+        {
+            IoTHubServiceClient = client;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            List<DeviceIdentity> deviceIdentities = new List<DeviceIdentity>();
+            for (int deviceIndex = 0; deviceIndex < BulkCount; deviceIndex++)
+            {
+                deviceIdentities.Add(
+                    new DeviceIdentity()
+                    {
+                        DeviceId = $"device_{deviceIndex}_{Random.Next(MaxRandomValue)}"
+                    });
+            }
+
+            // Create multiple DeviceIdentities.
+            await CreateDeviceIdentitiesAsync(deviceIdentities);
+
+            // Get the created devices from the service
+            IEnumerable<DeviceIdentity> createdIdentities = await GetDeviceIdentities(deviceIdentities.Select(identity => identity.DeviceId).ToArray()).ConfigureAwait(false);
+
+            // Update multiple DeviceIdentities.
+            await UpdateDeviceIdentitiesAsync(createdIdentities);
+
+            // Get the updated devices from the service
+            IEnumerable<TwinData> deviceTwins = await GetDeviceTwins(deviceIdentities.Select(identity => identity.DeviceId).ToArray()).ConfigureAwait(false);
+
+            // Update Device Twin.
+            await UpdateDeviceTwinsAsync(deviceTwins);
+
+            // Get the updated devices from the service
+            IEnumerable<DeviceIdentity> identitiesBeforeDelete = await GetDeviceIdentities(deviceIdentities.Select(identity => identity.DeviceId).ToArray()).ConfigureAwait(false);
```

Comment:
Why do we need this?

Reply:
So that we can delete these devices while respecting the ifMatch etag